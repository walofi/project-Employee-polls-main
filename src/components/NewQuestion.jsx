/* eslint-disable react/prop-types */
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import { connect } from "react-redux";
import { handleAddQuestion } from "../actions/questions";
import { useNavigate } from "react-router-dom";
import { useRef } from "react";

const NewQuestion = (props) => {
  const navigate = useNavigate();

  const firstOptionRef = useRef("");
  const secondOptionRef = useRef("");

  const handleSubmit = () => {
    localStorage.setItem("handleAnswer", true);
    if (
      firstOptionRef.current.value === "" ||
      secondOptionRef.current.value === ""
    ) {
      return alert("Please fill in all fields");
    }
    const question = {
      optionOneText: firstOptionRef.current.value,
      optionTwoText: secondOptionRef.current.value,
      author: props.authedUser,
    };
    props.dispatch(handleAddQuestion(question));
    navigate("/");
    firstOptionRef.current.value = "";
    secondOptionRef.current.value = "";
  };
  return (
    <Form className="text-center">
      <h2>Would You Rathar</h2>
      <p>Create Your Own Poll</p>
      <Form.Group className="mb-3" controlId="formFirstOption">
        <Form.Label data-testid="firstOptionLabel">First Option</Form.Label>
        <Form.Control
          data-testid="firstOption"
          ref={firstOptionRef}
          type="text"
          placeholder="First Option"
        />
      </Form.Group>

      <Form.Group className="mb-3" controlId="formFirstOption">
        <Form.Label data-testid="secondOptionLabel">Second Option</Form.Label>
        <Form.Control
          data-testid="secondOption"
          ref={secondOptionRef}
          type="text"
          placeholder="Second Option"
        />
      </Form.Group>

      <Button
        data-testid="submitQuestion"
        variant="primary"
        onClick={handleSubmit}
      >
        Submit
      </Button>
    </Form>
  );
};

const mapStateToProps = ({ authedUser }) => {
  return {
    authedUser,
  };
};

export default connect(mapStateToProps)(NewQuestion);
