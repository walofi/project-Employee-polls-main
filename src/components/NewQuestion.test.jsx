import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../store";
import { BrowserRouter } from "react-router-dom";
import React from "react";
import NewQuestion from "./NewQuestion";

describe("NewQuestion", () => {
  it("should render the component", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewQuestion />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });

  it("should display all elements", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewQuestion />
        </BrowserRouter>
      </Provider>
    );

    const firstOptionLabelElement = component.getByTestId("firstOptionLabel");
    const firstOptionInputElement = component.getByTestId("firstOption");
    const secondOptionLabelElement = component.getByTestId("secondOptionLabel");
    const secondOptionInputElement = component.getByTestId("secondOption");
    const submitButtonElement = component.getByTestId("submitQuestion");

    expect(firstOptionLabelElement.textContent).toBe("First Option");
    expect(secondOptionLabelElement.textContent).toBe("Second Option");
    expect(submitButtonElement.textContent).toBe("Submit");

    fireEvent.change(firstOptionInputElement, {
      target: { value: "have a pizza party" },
    });
    fireEvent.change(secondOptionInputElement, {
      target: { value: "have a promotion" },
    });
    expect(firstOptionInputElement.value).toBe("have a pizza party");
    expect(secondOptionInputElement.value).toBe("have a promotion");
  });
});
