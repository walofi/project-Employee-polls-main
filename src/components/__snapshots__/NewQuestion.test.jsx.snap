// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQuestion should render the component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="text-center"
      >
        <h2>
          Would You Rathar
        </h2>
        <p>
          Create Your Own Poll
        </p>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            data-testid="firstOptionLabel"
            for="formFirstOption"
          >
            First Option
          </label>
          <input
            class="form-control"
            data-testid="firstOption"
            id="formFirstOption"
            placeholder="First Option"
            type="text"
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            data-testid="secondOptionLabel"
            for="formFirstOption"
          >
            Second Option
          </label>
          <input
            class="form-control"
            data-testid="secondOption"
            id="formFirstOption"
            placeholder="Second Option"
            type="text"
          />
        </div>
        <button
          class="btn btn-primary"
          data-testid="submitQuestion"
          type="button"
        >
          Submit
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="text-center"
    >
      <h2>
        Would You Rathar
      </h2>
      <p>
        Create Your Own Poll
      </p>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          data-testid="firstOptionLabel"
          for="formFirstOption"
        >
          First Option
        </label>
        <input
          class="form-control"
          data-testid="firstOption"
          id="formFirstOption"
          placeholder="First Option"
          type="text"
        />
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          data-testid="secondOptionLabel"
          for="formFirstOption"
        >
          Second Option
        </label>
        <input
          class="form-control"
          data-testid="secondOption"
          id="formFirstOption"
          placeholder="Second Option"
          type="text"
        />
      </div>
      <button
        class="btn btn-primary"
        data-testid="submitQuestion"
        type="button"
      >
        Submit
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
